pkgs/development/compilers/gnu-smalltalk/default.nix

```diff
@@ -0,0 +1,50 @@
+{ stdenv, fetchurl, pkgconfig, libtool, zip, libffi, libsigsegv, readline, gmp,
+gnutls, gnome, cairo, SDL, sqlite }:
+
+let # The gnu-smalltalk project has a dependency to the libsigsegv library.
+    # The project ships with sources for this library, but deprecated this option.
+    # Using the vanilla libsigsegv library results in error: "cannot relocate [...]"
+    # Adding --enable-static=libsigsegv to the gnu-smalltalk configuration flags
+    # does not help, the error still occurs. The only solution is to build a
+    # shared version of libsigsegv.
+    libsigsegv-shared = stdenv.lib.overrideDerivation libsigsegv (oldAttrs: {
+      configureFlags = [ "--enable-shared" ];
+    });
+
+in stdenv.mkDerivation rec {
+
+  version = "3.2.5";
+  name = "gnu-smalltalk-${version}";
+
+  src = fetchurl {
+    url = "mirror://gnu/smalltalk/smalltalk-${version}.tar.xz";
+    sha256 = "1k2ssrapfzhngc7bg1zrnd9n2vyxp9c9m70byvsma6wapbvib6l1";
+  };
+
+  # The dependencies and their justification are explained at
+  # http://smalltalk.gnu.org/download
+  buildInputs = [
+    pkgconfig libtool zip libffi libsigsegv-shared readline gmp gnutls gnome.gtk
+    cairo SDL sqlite
+  ];
+
+  configureFlags = [ "--without-emacs" ];
```

Comment:
Unfortunately I have no idea how to make Emacs aware of where the gnu-smalltalk package is. Or is it simply adding emacs to the build inputs?